File: Core/src/main/java/com/github/ezauton/core/localization/estimators/CompassRotationalLocationEstimator.java
package com.github.ezauton.core.localization.estimators;

import com.github.ezauton.core.localization.RotationalLocationEstimator;
import com.github.ezauton.core.localization.sensors.Compass;

public class CompassRotationalLocationEstimator implements RotationalLocationEstimator {
    private final Compass compass;

    public CompassRotationalLocationEstimator(Compass compass) {
        this.compass = compass;
    }


    @Override
    public double estimateHeading() {
        return compass.getRadians();
    }
}

File: Core/src/main/kotlin/com/github/ezauton/core/localization/estimators/CompassRotationalLocationEstimator.kt
package com.github.ezauton.core.localization.estimators

import com.github.ezauton.core.localization.RotLocEst
import com.github.ezauton.core.localization.sensors.Compass

/**
 * TODO: fix... this seems redundant
 */
class CompassRotationalLocationEstimator(private val compass: Compass) : RotLocEst {

  override fun estimateHeading() = compass.angle

}
